Update of Existing OSIRIS-REx OCams Instrument Support

kledmundson opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail including motivation and any context -->
Updating existing support for OSIRIS-REx OCams instruments (MapCam, PolyCam, and SamCam) to reflect the current state in the U. of Arizona code base. Includes...

1. conversion of ocams2isis app to callable function
2. ctest for ocams2isis
3. conversion of ocams Makefile category tests to ctest module tests
4. deletion of existing ocams category Makefile tests
5. ocams fits files added to isis test data (2 each for poly, map, samcam)
6. updates to ocams camera fixture tests
7. ocams addendum kernels v10, v12 added to OsirisRexOcamsCamera/asset folder
8. Update tagcams serial number translation file to match that of ocams
    
## Related Issue
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/DOI-USGS/ISIS3/issues/5426

## How Has This Been Validated?
<!--- All changes need to be validated to confirm that they produce -->
<!--- scientifically accurate results. -->

Ocams Makefile category tests have been converted to ctest module tests with updated images and tests. All new and existing OSIRIS-REx tests passing.

ctest suite results...

Start 1164: TempTestingFiles.UnitTestOsirisRexTagCamsNAVCam
 1/23 Test #1164: TempTestingFiles.UnitTestOsirisRexTagCamsNAVCam ...........   Passed    1.27 sec
      Start 1165: TempTestingFiles.UnitTestOsirisRexTagCamsNFTCam
 2/23 Test #1165: TempTestingFiles.UnitTestOsirisRexTagCamsNFTCam ...........   Passed    0.89 sec
      Start 2294: OsirisRexMapCamModules.MapCamModuleExplodeReuniteTest
 3/23 Test #2294: OsirisRexMapCamModules.MapCamModuleExplodeReuniteTest .....   Passed   11.40 sec
      Start 2295: OsirisRexMapCamModules.MapCamModuleAlgebraTest
 4/23 Test #2295: OsirisRexMapCamModules.MapCamModuleAlgebraTest ............   Passed    1.11 sec
      Start 2296: OsirisRexMapCamModules.MapCamModuleInfoAndStatsTest
 5/23 Test #2296: OsirisRexMapCamModules.MapCamModuleInfoAndStatsTest .......   Passed   22.65 sec
      Start 2297: OsirisRexMapCamModules.MapCamModuleCamptTest
 6/23 Test #2297: OsirisRexMapCamModules.MapCamModuleCamptTest ..............   Passed    3.92 sec
      Start 2298: OsirisRexMapCamModules.MapCamModuleTwoImageTest
 7/23 Test #2298: OsirisRexMapCamModules.MapCamModuleTwoImageTest ...........   Passed   62.60 sec
      Start 2299: OsirisRexOcamsCube.PolyMath
 8/23 Test #2299: OsirisRexOcamsCube.PolyMath ...............................   Passed    1.56 sec
      Start 2300: OsirisRexOcamsCube.MappingCam
 9/23 Test #2300: OsirisRexOcamsCube.MappingCam .............................   Passed    1.56 sec
      Start 2301: OsirisRexOcamsCube.SamplingCam
10/23 Test #2301: OsirisRexOcamsCube.SamplingCam ............................   Passed    1.56 sec
      Start 2302: OsirisRexOcamsCube.PolyCamUpdatedIkCodes
11/23 Test #2302: OsirisRexOcamsCube.PolyCamUpdatedIkCodes ..................   Passed    1.55 sec
      Start 2303: OsirisRexPolyCamModules.PolyCamModuleExplodeReuniteTest
12/23 Test #2303: OsirisRexPolyCamModules.PolyCamModuleExplodeReuniteTest ...   Passed    9.42 sec
      Start 2304: OsirisRexPolyCamModules.PolyCamModuleAlgebraTest
13/23 Test #2304: OsirisRexPolyCamModules.PolyCamModuleAlgebraTest ..........   Passed    1.09 sec
      Start 2305: OsirisRexPolyCamModules.PolyCamModuleInfoAndStatsTest
14/23 Test #2305: OsirisRexPolyCamModules.PolyCamModuleInfoAndStatsTest .....   Passed   22.31 sec
      Start 2306: OsirisRexPolyCamModules.PolyCamModuleCamptTest
15/23 Test #2306: OsirisRexPolyCamModules.PolyCamModuleCamptTest ............   Passed    3.90 sec
      Start 2307: OsirisRexPolyCamModules.PolyCamModuleTwoImageTest
16/23 Test #2307: OsirisRexPolyCamModules.PolyCamModuleTwoImageTest .........   Passed   30.35 sec
      Start 2308: OsirisRexSamCamModules.SamCamModuleExplodeReuniteTest
17/23 Test #2308: OsirisRexSamCamModules.SamCamModuleExplodeReuniteTest .....   Passed    9.55 sec
      Start 2309: OsirisRexSamCamModules.SamCamModuleAlgebraTest
18/23 Test #2309: OsirisRexSamCamModules.SamCamModuleAlgebraTest ............   Passed    1.08 sec
      Start 2310: OsirisRexSamCamModules.SamCamModuleInfoAndStatsTest
19/23 Test #2310: OsirisRexSamCamModules.SamCamModuleInfoAndStatsTest .......   Passed   22.51 sec
      Start 2311: OsirisRexSamCamModules.SamCamModuleCamptTest
20/23 Test #2311: OsirisRexSamCamModules.SamCamModuleCamptTest ..............   Passed    3.91 sec
      Start 2312: OsirisRexSamCamModules.SamCamModuleTwoImageTest
21/23 Test #2312: OsirisRexSamCamModules.SamCamModuleTwoImageTest ...........   Passed   23.96 sec
      Start 2313: OsirisRexTagcamsNAVCamCube.NavigationCam
22/23 Test #2313: OsirisRexTagcamsNAVCamCube.NavigationCam ..................   Passed    2.80 sec
      Start 2314: OsirisRexTagcamsNFTCamCube.NaturalFeatureTrackingCam
23/23 Test #2314: OsirisRexTagcamsNFTCamCube.NaturalFeatureTrackingCam ......   Passed    2.77 sec

100% tests passed, 0 tests failed out of 23


<!--- If your changes include any new algorithms or changes to existing algorithms, -->
<!--- please indicate any publications or references for them. -->
<!--- If you manually validated the changes please indicate -->
<!--- what data you used and how you checked the results. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [X] My code follows the code style of this project. -->
- [X] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [X] I have added my user impacting change to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document. <!--- NOTE: You can only have one changelog entry per PR, see https://github.com/DOI-USGS/ISIS3/blob/dev/CONTRIBUTING.md -->


## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

KrisBecker committed:
Add OpenCV distortion Model to OCAMS. Addresses #5426.

* Added the OpenCV distortion model to the OSIRIS-REx OCAMS camera model. This distortion model is based upon the TAGCAMS OpenCV model implementation.

Signed-off-by: Kris J. Becker <kbecker@orex.lpl.arizona.edu>

KrisBecker committed:
Add new version of OCAMS distortion parameters

orex_ocams_addendum_v07.ti - Added new OCAMS distortion model parameters derived for 94 of the 117 PolyCam focus positions. With these new parameters, the OCAMS distortion algorithm is preferred over the OpenCV algorithm. It was installed in the ISIS3 processing system on 2019-09-04

Signed-off-by: Kris J. Becker <kbecker@orex.lpl.arizona.edu>

KrisBecker committed:
Updated focal lengths for OCAMS distortions

orex_ocams_addendum_v08.ti - The focal lenght updates for each of the 94 OCAMS distortion parameters where inadvertanty left out of the previous version v07. This update adds the individual INS-XXXXX_FOCAL_LENGTH parameter for each of the 94 focus positions. Installed 2019-10-21.

Signed-off-by: Kris J. Becker <kbecker@orex.lpl.arizona.edu>

KrisBecker committed:
Update to ORX OCAMS SamCam Distortion Mdle

orex_ocams_addendum_v09.ti - Add OCAMS SamCam distortion model developed by C. d’Aubigny (2020-02-17).

Signed-off-by: Kris J. Becker <kbecker@orex.lpl.arizona.edu>

kledmundson committed:
Update of existing support for OSIRIS-REx OCams instruments (MapCam, PolyCam, and SamCam) needs to reflect the current state in the U. of Arizona code base. Addresses #5426. Includes...
     1) conversion of ocams2isis app to callable function
     2) ctest for ocams2isis
     3) updates to OCamsCamera and OsirisRexDistortionMap
     4) conversion of ocams Makefile category tests to ctest module tests
     5) deletion of ocams category Makefile tests
     6) Addition of 2 ocams fits file each for poly, map, and samcam instruments to isis data area
     7) updates to ocams camera fixture tests

kledmundson committed:
Update to tagcams serial number translation file to match that of ocams. Addresses #5426.

kledmundson committed:
Added change log entry for ocams instrument support update. Addresses #5426.

kledmundson committed:
Merge branch 'DOI-USGS:dev' into enhancement/ocamsUpdate

acpaquette commented:
Hi @kledmundson, there are a few test failures in the build logs:
	785 - socetframesettings_app_test_orexMapCam (Failed)
	786 - socetframesettings_app_test_orexPolyCam (Failed)
	2131 - Phocube.FunctionalTestPhocubeSunIlluminationMask (Failed)
	2132 - Phocube.FunctionalTestPhocubeSurfaceObliqueDetectorResolution (Failed)
	2296 - OsirisRexMapCamModules.MapCamModuleTwoImageTest (Failed)
	2305 - OsirisRexPolyCamModules.PolyCamModuleTwoImageTest (Failed)
	2310 - OsirisRexSamCamModules.SamCamModuleTwoImageTest (Failed)
See [here](https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoiNDJNZ2MxbHFKTkwxV1RyQUxJekdJY3FIanNqU29rMHB4Nk1YUzk4REIrZUZDeEtEaW9HQlZ1dTZOSHpML2VUTGVDekYydmVFcU9sUHJKN20wQzd1Q0UzSzJscnB0MElDb1M3Ti9GTlJYR1RuMWJTV3V1SkJTa3NoYmc9PSIsIml2UGFyYW1ldGVyU3BlYyI6IjF3U2NTSGlDcEtCc29YVnEiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D/build/59775ba9-ed6a-4be3-9599-8b81d7936355) for more details

kledmundson commented:
Thanks, on it.

On Tue, Feb 25, 2025 at 15:23 Adam Paquette ***@***.***>
wrote:

> Hi @kledmundson <https://github.com/kledmundson>, there are a few test
> failures in the build logs:
> 785 - socetframesettings_app_test_orexMapCam (Failed)
> 786 - socetframesettings_app_test_orexPolyCam (Failed)
> 2131 - Phocube.FunctionalTestPhocubeSunIlluminationMask (Failed)
> 2132 - Phocube.FunctionalTestPhocubeSurfaceObliqueDetectorResolution
> (Failed)
> 2296 - OsirisRexMapCamModules.MapCamModuleTwoImageTest (Failed)
> 2305 - OsirisRexPolyCamModules.PolyCamModuleTwoImageTest (Failed)
> 2310 - OsirisRexSamCamModules.SamCamModuleTwoImageTest (Failed)
> See here
> <https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoiNDJNZ2MxbHFKTkwxV1RyQUxJekdJY3FIanNqU29rMHB4Nk1YUzk4REIrZUZDeEtEaW9HQlZ1dTZOSHpML2VUTGVDekYydmVFcU9sUHJKN20wQzd1Q0UzSzJscnB0MElDb1M3Ti9GTlJYR1RuMWJTV3V1SkJTa3NoYmc9PSIsIml2UGFyYW1ldGVyU3BlYyI6IjF3U2NTSGlDcEtCc29YVnEiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D/build/59775ba9-ed6a-4be3-9599-8b81d7936355>
> for more details
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DOI-USGS/ISIS3/pull/5718#issuecomment-2683429295>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABUGSUTS3RAM36LGPEFU32T2RTUN7AVCNFSM6AAAAABWEDZ2BCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOBTGQZDSMRZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: acpaquette]*acpaquette* left a comment (DOI-USGS/ISIS3#5718)
> <https://github.com/DOI-USGS/ISIS3/pull/5718#issuecomment-2683429295>
>
> Hi @kledmundson <https://github.com/kledmundson>, there are a few test
> failures in the build logs:
> 785 - socetframesettings_app_test_orexMapCam (Failed)
> 786 - socetframesettings_app_test_orexPolyCam (Failed)
> 2131 - Phocube.FunctionalTestPhocubeSunIlluminationMask (Failed)
> 2132 - Phocube.FunctionalTestPhocubeSurfaceObliqueDetectorResolution
> (Failed)
> 2296 - OsirisRexMapCamModules.MapCamModuleTwoImageTest (Failed)
> 2305 - OsirisRexPolyCamModules.PolyCamModuleTwoImageTest (Failed)
> 2310 - OsirisRexSamCamModules.SamCamModuleTwoImageTest (Failed)
> See here
> <https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoiNDJNZ2MxbHFKTkwxV1RyQUxJekdJY3FIanNqU29rMHB4Nk1YUzk4REIrZUZDeEtEaW9HQlZ1dTZOSHpML2VUTGVDekYydmVFcU9sUHJKN20wQzd1Q0UzSzJscnB0MElDb1M3Ti9GTlJYR1RuMWJTV3V1SkJTa3NoYmc9PSIsIml2UGFyYW1ldGVyU3BlYyI6IjF3U2NTSGlDcEtCc29YVnEiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D/build/59775ba9-ed6a-4be3-9599-8b81d7936355>
> for more details
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DOI-USGS/ISIS3/pull/5718#issuecomment-2683429295>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABUGSUTS3RAM36LGPEFU32T2RTUN7AVCNFSM6AAAAABWEDZ2BCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOBTGQZDSMRZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kledmundson committed:
Merge branch 'DOI-USGS:dev' into enhancement/ocamsUpdate

kledmundson committed:
Merge branch 'dev' into enhancement/ocamsUpdate

kledmundson committed:
Fixed test failures in ocams update branch. Addresses #5426.

kledmundson commented:
Think all failing tests are ok now...

**OsirisREx TwoImageTests**

Lowered test precision (they were too tight)

- OsirisRexPolyCamModules.PolyCamModuleTwoImageTest
- OsirisRexMapCamModules.MapCamModuleTwoImageTest
- OsirisRexSamCamModules.SamCamModuleTwoImageTest

**Phocube Tests**

Updated these tests and test data (isis/tests/data/phocube/20190425T211232S312_map_iofL2pan-crop.cub)

- Phocube.FunctionalTestPhocubeSunIlluminationMask
- Phocube.FunctionalTestSurfaceObliqueDetecterResolution

**Socetframesettings app tests for OSIRIS-REx MapCam & PolyCam**

- socetframesettings_app_test_orexMapCam

1. updated input to a more recent cube (20190223T023051S790_map_iofL2pan.cub)
2. updated truth data (sed_20190223T023051S790_map_iofL2pan_settings.txt)
3.  updated test Makefile  (isis/src/socet/apps/socetframesettings/tsts/orexMapCam/Makefile)

- socetframesettings_app_test_orexPolyCam

1. updated input cube by re-spiceinit (20190329T183101S0400_pol_iofL2pan.cub)
2. updated truth data (sed_20190329T183101S0400_pol_iofL2pan_settings.txt)
3. updated test Makefile (isis/src/socet/apps/socetframesettings/tsts/orexPolyCam/Makefile)

I've emailed the pertinent updated socetframesettings input and truth data to @acpaquette and @Kelvinrr

kledmundson committed:
Updated FunctionalTestCam2CamOffbody and test data in ocams update branch. Addresses #5426.

kledmundson commented:
One more failing test. Updated _FunctionalTestCam2CamOffbody_ and it's associated test data.

kledmundson committed:
Merge branch 'DOI-USGS:dev' into enhancement/ocamsUpdate

acpaquette commented on isis/src/osirisrex/objs/OsirisRexOcamsCamera/assets/orex_ocams_addendum_v05.ti:
```diff
@@ -0,0 +1,290 @@
+\begintext
```
@kledmundson Apologies for getting to this late, can these addendum files be removed? They are already in the ISIS data area so I am not sure why they are being included here

kledmundson commented:
yeah, removing now

On Thu, Apr 3, 2025 at 12:00 PM Adam Paquette ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> isis/src/osirisrex/objs/OsirisRexOcamsCamera/assets/orex_ocams_addendum_v05.ti
> <https://github.com/DOI-USGS/ISIS3/pull/5718#discussion_r2027579388>:
>
> > @@ -0,0 +1,290 @@
> +\begintext
>
> @kledmundson <https://github.com/kledmundson> Apologies for getting to
> this late, can these addendum files be removed? They are already in the
> ISIS data area so I am not sure why they are being included here
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DOI-USGS/ISIS3/pull/5718#pullrequestreview-2740873567>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABUGSURIA3R3Y776QNF2VUD2XWAN3AVCNFSM6AAAAABWEDZ2BCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDONBQHA3TGNJWG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kledmundson committed:
Merge branch 'DOI-USGS:dev' into enhancement/ocamsUpdate

kledmundson committed:
Removed ocams addendum files from ../isis/src/osirisrex/objs/OsirisRexOcamsCamera/assets/ folder. Addresses #5426.
